### ISSUE TITLE
Deprecate numbers at layers[].allowedAccounts[]

### ISSUE THREAD

**Issue by medikoo** [serverless/serverless#8305]
Currently we accept and eventually coerce a _number_ at account value:

https://github.com/serverless/serverless/blob/8c4d97211aa3dd4c41d9205a3ca0ccaab3564225/lib/plugins/aws/package/compile/layers/index.js#L77-L79

It's uncommon and doesn't follow general project conventions where we do not introduce such corecions.

It'll be great to clean this up, so first [deprecate](https://github.com/serverless/serverless/blob/master/CONTRIBUTING.md#configuring-deprecations) _number_ form, and with next major remove support for it.

**Comment by arjbholu** [comment]
Hi @medikoo 

I had like to work on this issue if that's okay. 

Correct me if I am wrong. Following is what I need to do:

1. I need to first add an entry to [deprecate](https://github.com/serverless/serverless/blob/master/CONTRIBUTING.md#configuring-deprecations) file.
2. Then add  `this.serverless._logDeprecation("code", "message")` to the part you mentioned in the comment. 

Apologies if there is any mistake. Please bear with me as this is my first time 😄 

Thanks,
Arjav

**Comment by medikoo** [comment]
@arjbholu great thanks for jumping in! Still today we've merged improvements that make this obsolete (sorry forgot to close it on time 🤦 ).

Through https://github.com/serverless/serverless/pull/8319 and #8299 users can input numbers and it'll be automatically coerced to string by validation engine (so in processing logic we can be sure we deal with strings).

Closing this now, to avoid further confusion
